### DISCUSSION TITLE
OMI glTF Working Group Meeting 2023-06-08

### DISCUSSION THREAD

**Discussion by robertlong** [omigroup/gltf-extensions#170] [General]
This meeting is on 5/25/2023 at 9:30 PM (UTC) / 2:30 PM (PST) in the [OMI Discord](https://discord.gg/NJtT9grz5E) within the #omi-weekly-meeting channel. During the meeting, we will be using the #omi-gltf-extensions channel to manage a speaker queue, post links, and for any sidebar discussions.

To be notified of this meeting and others, subscribe to the [OMI Meetings and Events Calendar](https://calendar.google.com/calendar/u/1?cid=Y18wZHB1Z2Y5ZjgzZXE0cWVrbWI2b21xYmptZ0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) or add yourself to the @omi-gltf-subgroup role in the #roles channel of the OMI Discord.

We have a bit of a problem due to DST, since the meeting time changes differently in different parts of the world

## Agenda:
* OMI Physics Extensions VS MSFT Extensions
  * https://github.com/omigroup/omigroup/discussions/393
  * Khronos Feedback
* Lightmap extension comparisons. Action items?
* OMI physics vehicles extension discussion
* OMI_physics_joint PRs:
  * https://github.com/omigroup/glTF-Validator/pull/2
  * https://github.com/omigroup/glTF-Validator/pull/3
  * https://github.com/omigroup/omi-godot/pull/1
* [Roblox glTF importer](https://devforum.roblox.com/t/3d-importer-support-for-gltf-files-beta/2320761)
  * [Outreach for OMI extensions](https://twitter.com/arobertlong/status/1656776528204816384)
* [EXT_manifold](https://github.com/KhronosGroup/glTF/pull/2286) 
* [OMI_personality quotes property](https://github.com/omigroup/gltf-extensions/issues/163)
* [KHR_audio comments](https://github.com/KhronosGroup/glTF/pull/2137#pullrequestreview-1412202653)
* [OMI_collider trimesh primitives](https://github.com/omigroup/gltf-extensions/issues/164)

## Backlog:
* OMI Vehicles Extensions Research
* OMI / MX Components Extension
* We discussed https://openmv.org (Open Metaverse Foundation) and how we can synergize with their efforts. Should we make a showcase of our projects to open up a broader discussion?
* jin - proposals how to make our specs and work more accessible to others. showcase gltf models
* Review pull requests https://github.com/omigroup/gltf-extensions/pulls
* [EXT_skin_humanoid](https://github.com/takahirox/EXT_skin_humanoid)
* [OMI_link](https://github.com/omigroup/gltf-extensions/tree/main/extensions/2.0/OMI_link)
* [OMI_spawn_point](https://github.com/omigroup/gltf-extensions/tree/main/extensions/2.0/OMI_spawn_point)
* Review [MSFT_physics](https://github.com/eoineoineoin/glTF_Physics)
  * [glTF_Physics_Blender_Exporter](https://github.com/eoineoineoin/glTF_Physics_Blender_Exporter)
  * [glTF_Physics_Godot_Importer](https://github.com/eoineoineoin/glTF_Physics_Godot_Importer)
* Licensing Extension https://twitter.com/superhoge/status/1596757861882740736
* [KHR_audio / Khronos feedback](https://github.com/KhronosGroup/glTF/pull/2137)
  * [gltf-sample-assets](https://github.com/KhronosGroup/glTF-Sample-Models/pull/360)
* [Third Room Extensions](https://github.com/matrix-org/thirdroom/pull/469)
* [glXF](https://github.com/KhronosGroup/glXF/blob/main/specification/2.0/README.md)
* Show and tell
* Create next weeks agenda

To propose another item for the agenda, comment below (preferably before the meeting).

**Comment by aaronfranke** [comment]
Discussed in call between Aaron and Robert: The MSFT approach to colliders by having the type be a key with separate sub-JSON would make it difficult to implement. An implementation could look for the first key, but then that would break in this case:

```json
{
    "extensions": {}
    "box": { "size": [1, 1, 1] }
}
```

**Comment by robertlong** [comment]
For generating Typescript types from the glTF spec's JSON Schema https://github.com/robertlong/gltf-typescript-generator
